api4/integration_action.go

```diff
@@ -123,3 +163,41 @@ func submitDialog(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write(b)
 }
+
+func executeIntegrationAction(c *Context, w http.ResponseWriter, r *http.Request) {
+	var action model.PluginIntegrationActionRequest
+
+	jsonErr := json.NewDecoder(r.Body).Decode(&action)
+	if jsonErr != nil {
+		c.SetInvalidParam("dialog")
+		return
+	}
+
+	if action.RequestURL == "" {
+		c.SetInvalidParam("url")
+		return
+	}
+
+	action.UserId = c.App.Session().UserId
+
+	if !c.App.SessionHasPermissionToChannel(*c.App.Session(), action.ChannelId, model.PERMISSION_READ_CHANNEL) {
```

Comment:
A bit out of the scope of the PR, but since you are adding the functionality...
We still need to define properly what is the "action context", and whether it always have a channel ID or team ID (for example, an item in the main menu may be independent of the channel or team). Therefore I would add to the condition something like `ChannelID != "" && ...`. Or maybe check on the location.